python/rmm/rmm.py

```diff
@@ -267,6 +267,43 @@ def get_ipc_handle(ary, stream=0):
     )
 
 
+try:
+    import cupy
+
+    class RMMCuPyMemory(cupy.cuda.memory.BaseMemory):
+        def __init__(self, size):
+            self.size = size
+            if size > 0:
+                self.rmm_array = librmm.device_buffer.DeviceBuffer(size=size)
+                self.ptr = self.rmm_array.ptr
+                self.device_id = cupy.cuda.runtime.pointerGetAttributes(
+                    self.ptr
+                ).device
+            else:
+                self.rmm_array = None
+                self.ptr = 0
+                self.device_id = cupy.cuda.device.get_device_id()
+
+
+except ImportError:
+    pass
```

Comment:
Would suggest doing...

```python
try:
     import cupy
except ImportError:
     cupy = None

if cupy:
   ...
```

That way we are not catching and suppressing other exceptions here.

Would also suggest adding `rmm_cupy_allocator` in this `if`.

Reply:
Good point, but I think we should always declare `rmm_cupy_allocator()`. 
It is better to get a `ImportError` exception so that users knows what went wrong. Else, we just going to have users asking us why `rmm.rmm_cupy_allocator()` isn't defined :)

Reply:
Makes sense :)

Reply:
> It is better to get a ImportError exception so that users knows what went wrong. 

Actually now I'm wondering if this doesn't just happen anyways since users would need to run `cupy.cuda.set_allocator(rmm.rmm_cupy_allocator)`, which would require running `import cupy` earlier. Or am I missing something?

Reply:
True, but on the off chance that someone calls `rmm.rmm_cupy_allocator()` directly or checking if `rmm.rmm_cupy_allocator` exist, I think it is best that `rmm.rmm_cupy_allocator` is always defined.